### PR TITLE
create installer inventory and copy it to the installer hosts

### PR DESCRIPTION
**Pull request by leena-jawale**
issue - https://issues.redhat.com/browse/AAP-18975

### CODE REVIEW

File: modules/db/variables.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 variable "resource_group" {
   description = "Azure resource group."
   type        = string
-  default     = "aap_on_azure"
+  default     = "leena-group"
```
Comment (chrisahl) [inline modules/db/variables.tf:4]: revert

File: main.tf [hunk 2]
```diff
@@ -14,7 +14,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "=3.83.0"
+      version = ">=3.83.0"
```
Comment (chrisahl) [inline main.tf:17]: should pin to specific version

File: modules/vm/variables.tf [hunk 3]
```diff
@@ -9,7 +9,7 @@ variable "deployment_id" {
 variable "resource_group" {
   description = "Azure resource group."
   type = string
-  default = "aap_on_azure"
+  default = "leena-group"
```
Comment (chrisahl) [inline modules/vm/variables.tf:12]: revert

File: main.tf [hunk 4]
```diff
@@ -138,6 +141,37 @@ module "eda" {
   deployment_id = var.deployment_id
   location = azurerm_resource_group.aap.location
   persistent_tags = local.persistent_tags
-  resource_group = azurerm_resource_group.aap.name
+  infrastructure_admin_username = var.infrastructure_admin_username
```
Comment (chrisahl) [inline main.tf:144]: Assume eda also needs resource_group

**Review by qjones7 — APPROVED** [review]

File: modules/vm/variables.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (leena-jawale) [reply modules/vm/variables.tf]: this was just used for testing and forgot to remove it

File: modules/vm/variables.tf [hunk 5]
```diff
@@ -71,7 +90,6 @@ variable "os_disk" {
 
 variable "infrastructure_admin_username" {
   type = string
-  default = "azureuser"
```
Comment (sharvarikhedkar) [inline modules/vm/variables.tf:74]: curious why this default value was removed?

Comment (leena-jawale) [reply modules/vm/variables.tf:74]: creating var for infrastructure_admin_username in root module https://github.com/ansible-content-lab/azure_terraform_deployment/pull/10/files#diff-05b5a57c136b6ff596500bcbfdcff145ef6cddea2a0e86d184d9daa9a65a288eR22-R27
and passing it https://github.com/ansible-content-lab/azure_terraform_deployment/pull/10/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR94 on each vm module, so no need to set default inside module

Comment (sharvarikhedkar) [reply modules/vm/variables.tf:74]: understood, thanks

**Review by sharvarikhedkar — APPROVED** [review]
From my testing I see `templates/inventory.j2` created as expected, lgtm

File: variables.tf [hunk 6]
```diff
@@ -84,3 +91,26 @@ variable "infrastructure_admin_ssh_private_key_filepath" {
   type = string
   default = "~/.ssh/id_rsa"
 }
+variable "aap_red_hat_username" {
+  description = "The RedHat Account name"
```
Comment (chrisahl) [inline variables.tf:95]: Change to "Red Hat account"

Do we need to be more descriptive of what account?

File: main.tf [hunk 7]
```diff
@@ -139,5 +142,37 @@ module "eda" {
...
+  provisioner "file" {
+      content = templatefile("${path.module}/templates/inventory.j2", { 
+        aap_controller_hosts = module.controller[*].nic_private_ip
+        aap_ee_hosts = module.execution[*].nic_private_ip
+        aap_hub_hosts = module.hub[*].nic_private_ip
+        aap_eda_hosts = module.eda[*].nic_private_ip
+        aap_eda_allowed_hostnames = module.eda[*].vm_public_ip
+        infrastructure_db_username = var.infrastructure_db_username
+        infrastructure_db_password = var.infrastructure_db_password
+        aap_red_hat_username = var.aap_red_hat_username
```
[the remark this answers is not in the document]
Comment (chrisahl) [reply main.tf:169]: Found out the aap_red_hat vars will be used in the near future to setup subscription manager on VMs

**Review by sharvarikhedkar — APPROVED** [review]

**Pull request merged**